src/scrot.c

```diff
@@ -285,33 +290,53 @@ static Imlib_Image scrotGrabAutoselect(void)
     return im;
 }
 
-/* similar to sleep, but deals with EINTR.
- * ideally, we'd want to sleep against an absolute time to prevent any drift
- * but clock_nanosleep doesn't seem to be widely implemented.
- */
-static void scrotSleep(int sec)
+void scrotDoDelay(void)
 {
-    assert(sec > 0);
-    struct timespec delay = { .tv_sec = sec };
-    while (nanosleep(&delay, &delay) < 0 && errno == EINTR);
+    if (!opt.delay)
```

Comment:
This should either be `opt.delay <= 0` or `opt.delay` should be initialized to 0 (currently it's initialized to -1, which makes this check useless):

```console
$ ./scrot -c
Taking shot in 0.
```

There was no `--delay` above, but the early return never worked and so the function is executed every time.